lib/AST/ASTDumper.cpp

```diff
@@ -609,8 +609,12 @@ namespace {
     void visitExtensionDecl(ExtensionDecl *ED) {
       printCommon(ED, "extension_decl", ExtensionColor);
       OS << ' ';
-      ED->getExtendedType().print(OS);
+      ED->getExtendedType()->getAnyNominal()->getDeclaredInterfaceType()->print(OS);
```

Comment:
We do not want to crash when dumping an extension of a non-nominal type, so this needs to be checked for, with a fallback. Also, you want `.print` (`->print` assumes the type is never null, which it could be).

Reply:
Ok, I tried to add a check but I'm not sure if this is what you meant. Could you give me an example of an extension on a non-nominal type type so I can test it?

Reply:
Yes, you got it right! 

> Could you give me an example of an extension on a non-nominal type so I can test it?

When the extended type is non-nominal, the extended type field is expected to contain an `ErrorType`. You can try something like `extension (Void, Never) {}`